### PR TITLE
Implement `encode_base64` and `decode_base64`

### PR DESCRIPTION
**Pull request by raxyte**
Implements functions:
- `encode_base64(blob|string) -> string`
- `decode_base64(blob|string) -> blob`

Closes https://github.com/tenzir/issues/issues/2360

### CODE REVIEW

File: web/docs/tql2/functions/decode_base64.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# decode_base64
+
+Decodes bytes as base64.
```
Comment (mavam) [inline web/docs/tql2/functions/decode_base64.md:3]:
```suggestion
Decodes bytes as Base64.
```

This is a named thing, so it's capitalized. Applies to all occurrences.

**Review by IyeOnline — APPROVED** [review]
GTG

File: changelog/next/features/4806--base64.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Two new functions `encode_base64` and `decode_base64` have been added to easily
+Base64 encode and decode blobs and strings.
```
Comment (dominiklohmann) [inline changelog/next/features/4806--base64.md:2]: Please prefer active voice and present tense in documentation and changelog entries.

File: libtenzir/builtins/functions/base64.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          check(b->Reserve(array.length()));
+          for (auto i = int64_t{}; i < array.length(); ++i) {
+            if (array.IsNull(i)) {
+              check(b->AppendNull());
+              continue;
+            }
+            if constexpr (Mode == mode::encode) {
+              check(b->Append(detail::base64::encode(array.Value(i))));
+            } else {
+              check(b->Append(detail::base64::decode(array.Value(i))));
```
[the remark this answers is not in the document]
Comment (raxyte) [reply libtenzir/builtins/functions/base64.cpp:54]: I was wondering why `decode` 's signature have no failure path. So it was a different function. Thanks!

File: changelog/next/features/4806--base64.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Add `encode_base64` and `decode_base64` functions to simplify Base64 encoding and
+decoding of blobs and strings.
```
[the remark this answers is not in the document]
Comment (jachris) [reply changelog/next/features/4806--base64.md:2]: Also, they don't really "simplify" it, they enable it from a TQL perspective (unless you consider the functionality Python provides, but I don't think that makes sense for us here).

**Review by IyeOnline — APPROVED** [review]

**Pull request merged**